var/spack/repos/builtin/packages/acl/package.py

```diff
@@ -23,14 +27,41 @@ class Acl(AutotoolsPackage):
     depends_on("autoconf", type="build")
     depends_on("automake", type="build")
     depends_on("libtool", type="build")
-    depends_on("attr")
     depends_on("gettext")
 
+    depends_on("attr")
+    # The configure step fails to find attr/xattr.h for these versions of acl unless this exact
+    # version of attr is used.
+    depends_on("attr@2.4.47", when="@:2.2.51")
+
+    # In 2.2.53 and later, the codebase provides a nice autogen script.
+    @when("@2.2.53:")
+    def autoreconf(self, spec, prefix):
+        autogen = Executable("./autogen.sh")
+        autogen()
+
+    # Before 2.2.53, the project's script to generate ./configure was contained in a Makefile which
+    # both generated the ./configure file and then executed a hardcoded ./configure command line
+    # with hardcoded install prefixes pointing to / and /usr. This patch removes the
+    # "make configure" target and replaces it with a "make autoconf" target that stops after
+    # generating the ./configure script, allowing spack to take over afterwards.
+    patch("makefile-autoconf-only.patch", when="@:2.2.52")
+
+    @when("@:2.2.52")
+    def autoreconf(self, spec, prefix):
+        make("autoconf")
+
+    # The "make configure" target in the unpatched Makefile touched this file after executing the
+    # hardcoded ./configure command line. The purpose of this file is not documented, and this step
+    # may not be necessary.
+    @when("@:2.2.52")
+    @run_after("configure")
+    def touch_census(self):
+        touch(".census")
```

Comment:
I'm pretty sure this is just to get the targets playing nicely in make, we shouldn't need it when we're not using `make configure` anymore. When you're running `configure`, there's not a file you know will be created by the configure script that you can key on to know whether it ran, but when you run `autoconf` you can key off the existence of the `configure` script. Nothing in the makefile appears to need it, so we should be fine getting rid of this

Reply:
Thank you, that makes perfect sense!